targets/unity-v2/source/Shared/Internal/Util.cs

```diff
@@ -74,16 +74,66 @@ public static string Format(string text, params object[] args)
         /// </summary>
         public static string ReadAllFileText(string filename)
         {
+            if (!File.Exists(filename))
+                return string.Empty;
+                
             if (_sb == null)
                 _sb = new StringBuilder();
             _sb.Length = 0;
 
-            var fs = new FileStream(filename, FileMode.Open);
-            var br = new BinaryReader(fs);
-            while (br.BaseStream.Position != br.BaseStream.Length)
-                _sb.Append(br.ReadChar());
+            FileStream fs = null;
+            try 
+            {
+                fs = new FileStream(filename, FileMode.Open);
+                using (BinaryReader br = new BinaryReader(fs))
+                {
+                    // Attempt to release the descriptor immediately
+                    fs = null;
 
+                    while (br.BaseStream.Position != br.BaseStream.Length)
+                        _sb.Append(br.ReadChar());
+                }
+            }
+            finally
+            {
+                // Make sure to dispose of the file stream in case the construction of binary reader fails
```

Comment:
Ah, I see the dispose here. If you just did a using for fs though, you wouldn't need the finally block, would you?

Reply:
We do `using` on `using(BinaryReader br = new BinaryReader(fs))` and in case that instantiation fails we want to make sure we still get rid of the file descriptor.

Reply:
Let's do `using` on both disposables actually.